tests/unit/query_model/test_nodes.py

```diff
@@ -443,3 +437,22 @@ def test_comparisons_between_value_nodes_are_strict():
     assert Value(10) != Value(10.0)
     assert Value(1) != Value(True)
     assert Value(10) != 10
+
+
+# TEST SORTING
+#
+
+
+def test_cannot_pick_row_from_unsorted_table():
+    events = SelectTable("events", EVENTS_SCHEMA)
+    with pytest.raises(TypeValidationError):
+        PickOneRowPerPatient(events, Position.FIRST)
+
+
+def test_can_pick_row_from_sorted_and_filtered_table():
+    events = SelectTable("events", EVENTS_SCHEMA)
+    date = SelectColumn(events, "date")
+    code = SelectColumn(events, "code")
+    sorted_by_date = Sort(events, date)
+    filtered_by_code = Filter(sorted_by_date, Function.EQ(code, Value("abc123")))
+    assert PickOneRowPerPatient(filtered_by_code, Position.FIRST)
```

Comment:
Is it worth having a test for the other way round too -- filter then sort?

Reply:
We cover that in a few places already e.g.
https://github.com/opensafely-core/databuilder/blob/fd05ad711afbb0edaffe5d80c3c494d39b6c932c/tests/unit/query_model/test_nodes.py#L299-L303

And
https://github.com/opensafely-core/databuilder/blob/fd05ad711afbb0edaffe5d80c3c494d39b6c932c/tests/unit/query_model/test_nodes.py#L51-L58

And possibly others, so I don't think we'd gain much here.